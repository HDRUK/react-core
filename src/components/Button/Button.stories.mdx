import { Meta, Story, Canvas } from "@storybook/addon-docs";
import SVG from "react-inlinesvg";

import Button from "./Button";
import { Card, CardBody } from "../Card";

import searchIcon from "../../images/icons/search.svg";
import userIcon from "../../images/icons/star.svg";

<Meta
  title="Molecules/Button"
  component={Button}
  parameters={{
    design: {
      type: "figma",
      url: "https://www.figma.com/file/fo4DadnYGfvtWawiQYcwmj/HDR-UK---Library?node-id=1021%3A16226",
    },
  }}
/>

<Canvas>
  <Story name="Variants">
    {(props) => (
      <Card>
        <CardBody>
          <div>
            <Button {...props} mb={4} mr={2}>
              Primary
            </Button>
            <Button {...props} mb={4} disabled>
              Primary (disabled)
            </Button>
          </div>
          <div>
            <Button {...props} variant="secondary" mb={4} mr={2}>
              Secondary
            </Button>
            <Button {...props} variant="secondary" mb={4} disabled>
              Secondary (disabled)
            </Button>
          </div>
          <div>
            <Button {...props} variant="tertiary" mr={2}>
              Tertiary
            </Button>
            <Button {...props} variant="tertiary" disabled>
              Tertiary (disabled)
            </Button>
          </div>
        </CardBody>
      </Card>
    )}
  </Story>
  <Story name="Sizes">
    {(props) => (
      <Card>
        <CardBody>
          <div>
            <Button {...props} mb={4} size="large">
              Large
            </Button>
          </div>
          <div>
            <Button {...props} mb={4}>
              Medium
            </Button>
          </div>
          <div>
            <Button {...props} size="small">
              Small
            </Button>
          </div>
        </CardBody>
      </Card>
    )}
  </Story>
  <Story name="Icons">
    {(props) => (
      <Card>
        <CardBody>
          <div>
            <Button {...props} mb={4} iconLeft={<SVG src={searchIcon} />}>
              Icon left
            </Button>
          </div>
          <div>
            <Button
              {...props}
              mb={4}
              iconRight={<SVG src={userIcon} fill="white" />}
            >
              Icon right
            </Button>
          </div>
          <div>
            <Button
              {...props}
              iconLeft={<SVG src={searchIcon} />}
              iconRight={<SVG src={userIcon} />}
            >
              Icon left and right
            </Button>
          </div>
        </CardBody>
      </Card>
    )}
  </Story>
</Canvas>
