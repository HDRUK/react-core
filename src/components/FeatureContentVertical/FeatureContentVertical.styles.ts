import { css, Theme } from "@emotion/react";
import { getComponentGlobals, getSize } from "../../theme";

export const root = (theme: Theme) => {
  const cardOffset = getComponentGlobals(
    "FeatureContentVertical",
    theme
  ).cardOffset;

  return css`
    margin: ${cardOffset};
  `;
};

export const media = (theme: Theme) => {
  return css`
    margin-bottom: ${getSize(2, theme)};

    > * {
      width: 100%;
    }
  `;
};
