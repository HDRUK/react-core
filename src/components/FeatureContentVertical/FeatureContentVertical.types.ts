import { FeatureContentProps } from "components/FeatureContent/FeatureContent.types";

export type FeatureContentVerticalProps = Omit<FeatureContentProps, "variant">;
