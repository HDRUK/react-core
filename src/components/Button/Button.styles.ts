import { css, Theme } from "@emotion/react";
import { getComponentSizeStyles, getComponentVariantStyles } from "../../theme";
import { getSize } from "../../theme";
import { ButtonStyleProps } from "./Button.types";

export const mixins = {
  button: () =>
    `
        border-width: 2px;
        border-style: solid;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        line-height: 1em;
        cursor: pointer;

        &:disabled {
          cursor: default;
        }
`,
};

export const root =
  ({ variant, size }: ButtonStyleProps) =>
  (theme: Theme) =>
    css`
      ${mixins.button()}
      ${getComponentVariantStyles("Button", variant, theme)}
      ${getComponentSizeStyles("Button", size, theme)}

      border-radius: 4px;
      gap: ${getSize(1.5, theme)};
    `;
