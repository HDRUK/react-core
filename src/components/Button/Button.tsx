/** @jsxImportSource @emotion/react */
import { jsx } from "@emotion/react";
import React from "react";
import { cx } from "@emotion/css";
import * as styles from "./Button.styles";
import useCommonStyles from "../../hooks/useCommonStyles";
import { ButtonProps } from "./index";

const Button = ({
  variant,
  children,
  size,
  m,
  mt,
  mb,
  ml,
  mr,
  mx,
  my,
  width,
  minWidth,
  maxWidth,
  iconLeft,
  iconRight,
  className,
  ...outerProps
}: ButtonProps) => {
  const commonStyles = useCommonStyles({
    mt,
    mr,
    ml,
    mb,
    m,
    mx,
    my,
    width,
    maxWidth,
    minWidth,
  });

  return (
    <button
      {...outerProps}
      css={[commonStyles, styles.root({ size, variant })]}
      className={cx("ui-Button", className)}
    >
      {iconLeft} {children} {iconRight}
    </button>
  );
};

Button.defaultProps = {
  size: "medium",
  variant: "primary",
  type: "button",
  disabled: false,
};

export default Button;
