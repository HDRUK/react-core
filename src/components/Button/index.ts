import { ReactElement } from "react";
import Button from "./Button";

export default Button;
