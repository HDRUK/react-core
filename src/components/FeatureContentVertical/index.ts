import FeatureContentVertical from "./FeatureContentVertical";

export default FeatureContentVertical;
