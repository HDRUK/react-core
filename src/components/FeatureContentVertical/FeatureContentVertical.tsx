/** @jsxImportSource @emotion/react */
import { jsx } from "@emotion/react";
import { Card, CardBody } from "components/Card";

import * as styles from "./FeatureContentVertical.styles";
import { FeatureContentVerticalProps } from "./FeatureContentVertical.types";
import FeatureContentBody from "../FeatureContent/FeatureContentBody";
import FeatureContentHeader from "../FeatureContent/FeatureContentHeader";
import FeatureContentActions from "components/FeatureContent/FeatureContentActions";

const FeatureContentVertical = ({
  media,
  header,
  body,
  actions,
  ...outerProps
}: FeatureContentVerticalProps) => {
  return (
    <Card {...outerProps} maxWidth="284px">
      <CardBody>
        <div css={styles.root}>
          {header && (
            <FeatureContentHeader mb={3}>{header}</FeatureContentHeader>
          )}
          <div css={styles.media}>{media}</div>
          {body && <FeatureContentBody mb={4}>{body}</FeatureContentBody>}
          {actions && (
            <FeatureContentActions textAlign="center">
              {actions}
            </FeatureContentActions>
          )}
        </div>
      </CardBody>
    </Card>
  );
};

export default FeatureContentVertical;
