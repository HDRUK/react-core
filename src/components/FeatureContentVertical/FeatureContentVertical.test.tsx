import { RenderResult, screen } from "@testing-library/react";
import React from "react";
import { renderWithProviders } from "../../testUtils";
import Button from "../Button";
import FeatureContentVertical from "../FeatureContentVertical";

let wrapper: RenderResult;

describe("Given the FeatureContentVertical component", () => {
  describe("When it is rendered", () => {
    beforeEach(() => {
      wrapper = renderWithProviders(
        <FeatureContentVertical
          header="Header"
          body="Body"
          actions={<Button>View</Button>}
          media="Image"
        />
      );
    });

    it("Then should match the snapshot", async () => {
      expect(wrapper.asFragment()).toMatchSnapshot();
    });

    it("Then should have the correct header", async () => {
      const header = await screen.findByText(/Header/);

      expect(header).toBeTruthy();
    });

    it("Then should have the correct body", async () => {
      const body = await screen.findByText(/Body/);

      expect(body).toBeTruthy();
    });

    it("Then should have the correct actions", async () => {
      const actions = await screen.findByText(/View/);

      expect(actions).toBeTruthy();
    });

    it("Then should containg the correct image", async () => {
      const media = await screen.findByText(/Image/);

      expect(media).toBeTruthy();
    });
  });
});
