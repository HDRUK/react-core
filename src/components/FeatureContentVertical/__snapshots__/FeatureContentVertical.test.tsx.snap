// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the FeatureContentVertical component When it is rendered Then should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui-FeatureContentVertical ui-Card css-190dtii-Card"
    role="article"
  >
    <div
      class="ui-CardBody css-4iti4b-CardBody"
    >
      <div
        class="css-1g4h4q5-FeatureContentVertical"
      >
        <h5
          class="ui-Typography css-yngmzh-Typography"
        >
          <div
            class="ui-Box css-12ou0ad-Box"
          >
            Header
          </div>
        </h5>
        <div
          class="css-wvev0n-FeatureContentVertical"
        >
          Image
        </div>
        <div
          class="ui-Box css-30redr-Box"
        >
          Body
        </div>
        <div
          class="ui-Box css-1m03czi-Box"
        >
          <button
            class="ui-Button css-zdx1uk-Button"
            type="button"
          >
             View 
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
